dynatrace/api/v2/settings/objects/permissions/service_test.go

```diff
@@ -132,6 +162,68 @@ func TestService(t *testing.T) {
 		})
 	})
 
+	t.Run("List", func(t *testing.T) {
+		t.Run("Returns stubs for all object IDs", func(t *testing.T) {
+			settingsClient := &settingsClientStub{
+				getSchemaIDsWithOwnerBasedAccessControl: func(ctx context.Context) ([]string, error) {
+					return []string{"schemaID1", "schemaID2"}, nil
+				},
+				listObjectsIDsOfSchema: func(ctx context.Context, schemaID string) ([]string, error) {
+					if schemaID == "schemaID1" {
+						return []string{"objectID1", "objectID2"}, nil
+					}
+					if schemaID == "schemaID2" {
+						return []string{"objectID3"}, nil
+					}
+					return nil, nil
+				},
+			}
+			service := permissionService.ServiceImpl{SettingsClient: settingsClient}
+			stubs, err := service.List(t.Context())
+			require.NoError(t, err)
+			assert.Len(t, stubs, 3)
+			assert.Equal(t, &api.Stub{ID: "objectID1", Name: "objectID1"}, stubs[0])
+			assert.Equal(t, &api.Stub{ID: "objectID2", Name: "objectID2"}, stubs[1])
+			assert.Equal(t, &api.Stub{ID: "objectID3", Name: "objectID3"}, stubs[2])
+		})
+
+		t.Run("Returns error if GetSchemaIDsWithOwnerBasedAccessControl fails", func(t *testing.T) {
+			settingsClient := &settingsClientStub{
+				getSchemaIDsWithOwnerBasedAccessControl: func(ctx context.Context) ([]string, error) {
+					return nil, assert.AnError
+				},
+			}
+			service := permissionService.ServiceImpl{SettingsClient: settingsClient}
+			stubs, err := service.List(t.Context())
+			assert.Error(t, err)
+			assert.Nil(t, stubs)
+		})
+
+		t.Run("Returns error if ListObjectsIDsOfSchema fails", func(t *testing.T) {
+			settingsClient := &settingsClientStub{
+				getSchemaIDsWithOwnerBasedAccessControl: func(ctx context.Context) ([]string, error) {
+					return []string{"schemaID1"}, nil
+				},
+				listObjectsIDsOfSchema: func(ctx context.Context, schemaID string) ([]string, error) {
+					return nil, assert.AnError
+				},
+			}
+			service := permissionService.ServiceImpl{SettingsClient: settingsClient}
+			stubs, err := service.List(t.Context())
+			assert.Error(t, err)
+			assert.Nil(t, stubs)
+		})
+
+		t.Run("Returns error if getSettingsClient fails", func(t *testing.T) {
+			service := permissionService.ServiceImpl{
+				Credentials: &rest.Credentials{},
+			}
+			stubs, err := service.List(t.Context())
+			assert.Error(t, err)
```

Comment:
[nit] We could also validate the error here `assert.ErrorIs(t, rest.NoPlatformCredentialsErr, err)`